repose-aggregator/components/filters/add-header/src/test/scala/org/openrepose/filters/addheader/AddHeaderFilterTest.scala

```diff
@@ -0,0 +1,222 @@
+
+      servletResponse.containsHeader("x-new-header-1") shouldBe true
+      servletResponse.getHeader("x-new-header-1") should include ("new-value-1;q=0.2")
+    }
+
+    it("should contain added header without removing original") {
+      filter.configurationUpdated(addHeaderResponseConfig(removeOriginal = false))
+      addResponseHeaders(List(SimpleHeader("x-new-header-1", "old-value-1")))
+
+      filter.doFilter(servletRequest, servletResponse, filterChain)
+
+      servletResponse.containsHeader("x-new-header-1") shouldBe true
+      servletResponse.getHeaders("x-new-header-1") should contain theSameElementsAs List("old-value-1", "new-value-1;q=0.2")
+    }
+
+    it("should contain added header after removing original") {
+      filter.configurationUpdated(addHeaderResponseConfig(removeOriginal = true))
+      addResponseHeaders(List(SimpleHeader("x-new-header-1", "old-value-1"), SimpleHeader("other-header", "some-value")))
+
+      filter.doFilter(servletRequest, servletResponse, filterChain)
+
+      servletResponse.containsHeader("x-new-header-1") shouldBe true
+      servletResponse.getHeaders("x-new-header-1") should have size 1
+      servletResponse.getHeader("x-new-header-1") should include ("new-value-1;q=0.2")
+      servletResponse.getHeader("x-new-header-1") should not include "old-value-1"
+      servletResponse.containsHeader("other-header") shouldBe true  // ensure our mock-magic is working
+    }
+
+    it("should contain added header with multiple values") {
+      filter.configurationUpdated(addHeaderResponseConfig(removeOriginal = false, numValues = 3))
+
+      filter.doFilter(servletRequest, servletResponse, filterChain)
+
+      servletResponse.containsHeader("x-new-header-1") shouldBe true
+      servletResponse.getHeaders("x-new-header-1") should contain theSameElementsAs List("new-value-1;q=0.2", "new-value-2;q=0.2", "new-value-3;q=0.2")
+    }
+
+    it("should contain multiple added headers with multiple values") {
+      filter.configurationUpdated(addHeaderResponseConfig(removeOriginal = false, numValues = 3, numHeaders = 3))
+
+      filter.doFilter(servletRequest, servletResponse, filterChain)
+
+      servletResponse.containsHeader("x-new-header-1") shouldBe true
+      servletResponse.getHeaders("x-new-header-1") should contain theSameElementsAs List("new-value-1;q=0.2", "new-value-2;q=0.2", "new-value-3;q=0.2")
+      servletResponse.containsHeader("x-new-header-2") shouldBe true
+      servletResponse.getHeaders("x-new-header-2") should contain theSameElementsAs List("new-value-4;q=0.2", "new-value-5;q=0.2", "new-value-6;q=0.2")
+    }
+  }
+
+  def getPostFilterRequest: HttpServletRequestWrapper = {
+    val requestCaptor = ArgumentCaptor.forClass(classOf[HttpServletRequestWrapper])
+    verify(filterChain).doFilter(requestCaptor.capture(), any(classOf[ServletResponse]))
+    requestCaptor.getValue
+  }
+
+  def addResponseHeaders(headers: Iterable[SimpleHeader]): Unit = {
```

Comment:
Hmmm, maybe not because you need some magic to happen in the middle. I forgot about that.